conans/test/util/detect_test.py

```diff
@@ -1,15 +1,60 @@
+import platform
 import unittest
-from conans.test.utils.tools import TestBufferConanOutput
+from subprocess import Popen, PIPE, STDOUT
+
+from conans import tools
 from conans.client.conf.detect import detect_defaults_settings
-import platform
+from conans.test.utils.tools import TestBufferConanOutput
 
 
 class DetectTest(unittest.TestCase):
 
-    def detect_test(self):
+    def detect_default_compilers_test(self):
+        platform_default_compilers = {
+            "Linux": "gcc",
+            "Darwin": "apple-clang",
+            "Windows": "Visual Studio"
+        }
         output = TestBufferConanOutput()
         result = detect_defaults_settings(output)
-        if platform.system() == "Linux":
-            for (name, value) in result:
-                if name == "compiler":
-                    self.assertEquals(value, "gcc")
+        # result is a list of tuples (name, value) so converting it to dict
+        result = dict(result)
+        platform_compiler = platform_default_compilers.get(platform.system(), None)
+        if platform_compiler is not None:
+            self.assertEquals(result.get("compiler", None), platform_compiler)
+
+    def detect_default_in_mac_os_using_gcc_as_default_test(self):
+        """
+        Test if gcc in Mac OS X is using apple-clang as frontend
+        """
+        # See: https://github.com/conan-io/conan/issues/2231
+        if platform.system() != "Darwin":
+            return
+
+        def _execute(command):
```

Comment:
You can simplify all this with `output_str = subprocess.check_output(command)` and try, except in case of subprocess failure.

Reply:
Yes, totally right!!! fixing it!!!